### PR TITLE
Development

### PR DESCRIPTION
**Pull request by shloch**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    t.integer "post_id"
+  end
+
+  create_table "posts", force: :cascade do |t|
+    t.string "title"
+    t.string "url"
+    t.string "category"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "user_id"
```
Comment (rvvergara) [inline db/schema.rb:30]: We need to add a `foreign_key` constraint for `user_id` here :) For the what, why and how please see [this](https://thoughtbot.com/blog/referential-integrity-with-foreign-keys)

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2019_05_15_093917) do
+
+  create_table "comments", force: :cascade do |t|
+    t.string "text"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.integer "user_id"
+    t.integer "post_id"
```
Comment (rvvergara) [inline db/schema.rb:20]: Let's include a `foreign_key` constraint here for `user_id` and `post_id`.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+    belongs_to  :user
+    has_many  :comments
+
+    validates :title,  presence: true, length: { minimum: 10 }
+    validates :category,  length: { minimum: 3 }
+    VALID_URL_REGEX = /(http|https):\/\/[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?/i 
+      validates :url, presence: true, length: { maximum: 255 },
```
Comment (rvvergara) [inline app/models/post.rb]: Why did we need to validate url for `post`? And in fact, why do we need this attribute at all? :)

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+    belongs_to :user
+    belongs_to :post
+
+      validates :text,  presence: true, length: { minimum: 6 }
+      validates :user_id,  presence: true
```
Comment (rvvergara) [inline app/models/comment.rb]: The validation for the user_id and post_id are both handled by the associations in lines 2-3 :) Without these two lines 6-7 we still will prevent a comment creation without user_id and post_id

**Review by rvvergara — CHANGES_REQUESTED** [review]
Hi @shloch,

Great work here! I tested it in my console and everything works just fine. Though to polish the code I just have some suggested changes. You may see them in my specific comments :)

File: db/schema.rb (hunk 1 above)
Comment (shloch) [reply db/schema.rb:30]: Thanks for observation.... this has been added

File: app/models/post.rb (hunk 3 above)
Comment (shloch) [reply app/models/post.rb]: It has been deleted

File: app/models/comment.rb (hunk 4 above)
Comment (shloch) [reply app/models/comment.rb]: Thanks for observations

**Review by rvvergara — APPROVED** [review]
Hi bro. The foreign_key constraint is still missing. I would expect to see it in the last few lines of the schema, something like:

`add_foreign_key "posts", "users", column: "user_id"`

I saw you added migrations to do this, but it appears the necessary foreign_key constraints weren't included. 

At any rate since you satisfied all the other necessary requirements I will just leave this as a homework to you. But I believe you already deserve a

# :+1: 

PS: For your succeeding projects so that the foreign_key constraints would automatically be included just indicated `user:references` when you are for example creating a model that has `user_id` as one of its columns.

**Pull request merged**